File: tests/src/test/java/org/veri/be/integration/support/stub/StubStorageService.java
package org.veri.be.integration.support.stub;

import org.veri.be.global.storage.dto.PresignedPostFormResponse;
import org.veri.be.global.storage.dto.PresignedUrlResponse;
import org.veri.be.global.storage.service.StorageService;

import java.time.Duration;

public class StubStorageService implements StorageService {
    @Override
    public PresignedUrlResponse generatePresignedUrl(String contentType, long contentLength, String prefix, Duration duration) {
        return new PresignedUrlResponse("http://stub.presigned.url", null);
    }

    @Override
    public PresignedUrlResponse generatePresignedUrlOfDefault(String contentType, long contentLength) {
        return new PresignedUrlResponse("http://stub.presigned.url", null);
    }

    @Override
    public PresignedPostFormResponse generatePresignedPost(String contentType, long fileSize, String prefix, Duration duration) {
        return new PresignedPostFormResponse("http://stub.s3.url", java.util.Map.of("key", "value"));
    }
}

File: tests/src/test/kotlin/org/veri/be/integration/support/stub/StubStorageService.kt
package org.veri.be.integration.support.stub

import org.veri.be.global.storage.dto.PresignedPostFormResponse
import org.veri.be.global.storage.dto.PresignedUrlResponse
import org.veri.be.global.storage.service.StorageService
import java.time.Duration

class StubStorageService : StorageService {
    override fun generatePresignedUrl(
        contentType: String,
        contentLength: Long,
        prefix: String,
        duration: Duration
    ): PresignedUrlResponse = PresignedUrlResponse("http://stub.presigned.url", null)

    override fun generatePresignedUrlOfDefault(
        contentType: String,
        contentLength: Long
    ): PresignedUrlResponse = PresignedUrlResponse("http://stub.presigned.url", null)

    override fun generatePresignedPost(
        contentType: String,
        fileSize: Long,
        prefix: String,
        duration: Duration
    ): PresignedPostFormResponse =
        PresignedPostFormResponse("http://stub.s3.url", mapOf("key" to "value"))
}
